editor/plugins/navigation_polygon_editor_plugin.cpp

```diff
@@ -198,14 +200,53 @@ void NavigationPolygonEditor::_bake_pressed() {
 	button_bake->set_pressed(false);
 
 	ERR_FAIL_NULL(node);
-	Ref<NavigationPolygon> navigation_polygon = node->get_navigation_polygon();
-	if (navigation_polygon.is_null()) {
+	Ref<NavigationPolygon> navigation_mesh = node->get_navigation_polygon();
+	if (navigation_mesh.is_null()) {
 		err_dialog->set_text(TTR("A NavigationPolygon resource must be set or created for this node to work."));
 		err_dialog->popup_centered();
 		return;
 	}
 
-	node->bake_navigation_polygon(true);
+	Vector<Vector2> undo_vertices = navigation_mesh->get_vertices();
+	Vector<Vector<int>> undo_polygon_indices;
+	int undo_polygon_count = navigation_mesh->get_polygon_count();
+	undo_polygon_indices.resize(undo_polygon_count);
+	for (int i = 0; i < undo_polygon_count; i++) {
+		undo_polygon_indices.write[i] = navigation_mesh->get_polygon(i);
+	}
+
+	Ref<NavigationMeshSourceGeometryData2D> source_geometry_data;
+	source_geometry_data.instantiate();
+
+	NavigationServer2D::get_singleton()->parse_source_geometry_data(navigation_mesh, source_geometry_data, node);
+	NavigationServer2D::get_singleton()->bake_from_source_geometry_data(navigation_mesh, source_geometry_data);
+
+	Vector<Vector2> redo_vertices = navigation_mesh->get_vertices();
+	Vector<Vector<int>> redo_polygon_indices;
+	int redo_polygon_count = navigation_mesh->get_polygon_count();
+	redo_polygon_indices.resize(redo_polygon_count);
+	for (int i = 0; i < redo_polygon_count; i++) {
+		redo_polygon_indices.write[i] = navigation_mesh->get_polygon(i);
+	}
```

Comment:
this block is the same for undo/redo indices/vertices so perhaps we can extract it to helper function to reduce duplication and increase readability a bit?

Reply:
When https://github.com/godotengine/godot/pull/93171 gets merged that polygon loop block would become unnecessary as it can be replaced by a simple `get_polygons()` function.